source/write/transactions.txt

```diff
@@ -0,0 +1,194 @@
+.. _cpp-transactions:
+
+============
+Transactions
+============
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: code example, rollback, undo operation
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to use the {+driver-long+} to perform
+**transactions**. Transactions allow
+you to run a series of operations that do not change any data until the
+transaction is committed. If any operation in the transaction returns an
+error, the driver cancels the transaction and discards all data changes
+before they ever become visible.
+
+In MongoDB, transactions run within logical **sessions**. A
+session is a grouping of related read or write operations that you intend to run sequentially. 
+Sessions enable causal consistency for a group of operations in an  **ACID-compliant**  transaction, which is a 
```

Comment:
For causal consistency, I do think it can be an unfamiliar term so I think it would be good to provide some sort of reference (doesn't have to be a link - could be a short definition). I don't feel too strongly about it though, so I defer to your judgement!